src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java

```diff
@@ -0,0 +1,96 @@
+package org.kiwiproject.dropwizard.poller.health;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static org.kiwiproject.base.KiwiStrings.format;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newUnhealthyResultBuilder;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.common.annotations.VisibleForTesting;
+import lombok.experimental.UtilityClass;
+import org.kiwiproject.dropwizard.poller.metrics.ClientPollerStatistics;
+import org.kiwiproject.metrics.health.HealthStatus;
+
+import java.net.URI;
+import java.time.temporal.TemporalUnit;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+@UtilityClass
+public class ClientPollerHealthChecks {
+
+    private static final String UNSPECIFIED_URI_PREFIX = "[unspecified-URI-";
+    private static final String UNSPECIFIED_URI_SUFFIX = "]";
+    private static final Supplier<String> NAME_FOR_UNSPECIFIED_URI = ClientPollerHealthChecks::nameForUnspecifiedUri;
+
+    @VisibleForTesting
+    static final String FAILURE_DETAILS_KEY = "FailureDetails";
+
+    private static String nameForUnspecifiedUri() {
+        return UNSPECIFIED_URI_PREFIX + System.currentTimeMillis() + UNSPECIFIED_URI_SUFFIX;
+    }
+
+    static String humanReadableOf(TemporalUnit unit) {
+        return unit.toString().toLowerCase(Locale.getDefault());
+    }
+
+    /**
+     * Build a name for the given {@code healthCheckType} and {@code pollingUri}
+     *
+     * @param healthCheckType The type of the health check
+     * @param pollingUri      The polling URI
+     * @return a name to represent this health check
+     */
+    public static String nameFor(String healthCheckType, URI pollingUri) {
+        var name = Optional.ofNullable(pollingUri).map(URI::toString).orElseGet(NAME_FOR_UNSPECIFIED_URI);
+        return prefixedNameFor(healthCheckType, name);
+    }
+
+    /**
+     * Build a name for the given {@code healthCheckType} and {@code pollingUri}
+     *
+     * @param healthCheckType The type of the health check
+     * @param pollingUri      The polling URI as a String
+     * @return a name to represent this health check
+     */
+    public static String nameFor(String healthCheckType, String pollingUri) {
+        var name = Optional.ofNullable(pollingUri).orElseGet(NAME_FOR_UNSPECIFIED_URI);
+        return prefixedNameFor(healthCheckType, name);
+    }
+
+    private static String prefixedNameFor(String healthCheckType, String resolvedUri) {
+        return format("client-poller-{}:{}", healthCheckType, resolvedUri);
+    }
+
+    /**
+     * Build an unhealthy {@link com.codahale.metrics.health.HealthCheck.Result} using the given statistics,
```

Comment:
You can replace the FQCN with the qualified one (HealthCheck.Result) here and the other places in this class

Reply:
You can use an IntelliJ intention (alt + enter) to do it for you...